internal/handlers/pg/msg_aggregate.go

```diff
@@ -99,39 +106,50 @@ func (h *Handler) MsgAggregate(ctx context.Context, msg *wire.OpMsg) (*wire.OpMs
 		}
 
 		var s aggregations.Stage
+
 		if s, err = aggregations.NewStage(d); err != nil {
 			return nil, err
 		}
 
-		stages[i] = s
+		switch s.Type() {
+		case aggregations.StageTypeDocuments:
+			stagesDocuments = append(stagesDocuments, s)
+		case aggregations.StageTypeStats:
+			stagesStats = append(stagesStats, s)
```

Comment:
This approach would work. I learnt from using `task run-proxy` with `mongosh` that $collStats has to be the first stage of pipeline. And if there are any stage after `$collStats`, then `$collStats` will be ignored. Anyway error handling is not part of this task so this can be just to do.

`task run-proxy` with `mongosh`
```
test> db.test.aggregate([{$count:"before"},{$collStats:{count:{}, storageStats:{}}},{$count:"after"}])
MongoServerError: $collStats is only valid as the first stage in a pipeline
test> db.test.aggregate([{$collStats:{count:{}, storageStats:{}}},{$count:"after"}])
[ { after: 1 } ]
test> 
```

`task run` with `mongosh`
```
test> db.test.aggregate([{$count:"before"},{$collStats:{count:{}, storageStats:{}}},{$count:"after"}])
[
  {
    ns: 'test.test',
    host: 'alma',
    localTime: '2023-04-03T02:55:14Z',
    storageStats: {
      size: Long("32768"),
      count: 2,
      avgObjSize: 4096,
      storageSize: Long("8192"),
      freeStorageSize: 0,
      capped: false,
      wiredTiger: {},
      nindexes: 1,
      indexDetails: {},
      indexBuilds: {},
      totalIndexSize: Long("16384"),
      indexSizes: {}
    },
    count: 2
  }
]
test> db.test.aggregate([{$collStats:{count:{}, storageStats:{}}},{$count:"after"}])
[
  {
    ns: 'test.test',
    host: 'alma',
    localTime: '2023-04-03T02:55:16Z',
    storageStats: {
      size: Long("32768"),
      count: 2,
      avgObjSize: 4096,
      storageSize: Long("8192"),
      freeStorageSize: 0,
      capped: false,
      wiredTiger: {},
      nindexes: 1,
      indexDetails: {},
      indexBuilds: {},
      totalIndexSize: Long("16384"),
      indexSizes: {}
    },
    count: 2
  }
]
```

Reply:
Ah, thanks a lot! This is very helpful for testing!

Reply:
Fixed in pair programming, more tests to be added in #2349